# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [4]:
# 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.

# Passo 1: Dividir o intervalo
# Já definimos n = 1000
import numpy as np

# Definindo a função
def f(x):
    return x**3


n = 1000

# Limites de integração
a, b = 1, 2

# Passo 2: Largura dos Retângulos (dx)
dx = (b - a) / n

# Coordenadas x dos retângulos
x = np.linspace(a + dx/2, b - dx/2, n)

# Área total usando o método dos retângulos
area_numerica = np.sum(f(x) * dx)
area_numerica, n


(3.7499996249999996, 1000)

In [5]:
# 2- Calcule essa área de forma analítica.

Para calcular a área sob a curva da função $f(x) = x^3$ no intervalo de 1 a 2, eu farei primeiro o cálculo numérico usando o método dos retângulos e depois o cálculo analítico.

### Cálculo Analítico

Para o cálculo analítico, integraremos a função $f(x) = x^3$ de 1 a 2. A integral indefinida de $x^3$ é $\frac{1}{4}x^4 $, e aplicaremos os limites de integração para encontrar a área exata.

Primeiro, vou calcular a área usando o método dos retângulos.

A integral definida de $f(x) = x^3$ de 1 a 2 é dada por:

$$\int_{1}^{2} x^3 \, dx = \left[ \frac{1}{4}x^4 \right]_{1}^{2} \rightarrow \frac{15}{4} \rightarrow 3.75$$

Portanto, o cálculo analítico resulta no mesmo valor, 3.75, para a área sob a curva no intervalo especificado.